compiler/src/dmd/safe.d

```diff
@@ -376,7 +377,15 @@ extern (D) void reportSafeError(FuncDeclaration fd, bool gag, Loc loc,
     else if (fd.isSafe() || fd.isSaferD())
     {
         if (!gag && format)
-            .error(loc, format, arg0 ? arg0.toChars() : "", arg1 ? arg1.toChars() : "", arg2 ? arg2.toChars() : "");
+        {
+            OutBuffer buf;
+            buf.printf(format, arg0 ? arg0.toChars() : "", arg1 ? arg1.toChars() : "", arg2 ? arg2.toChars() : "");
+            if (fd.isSafe())
+                buf.writestring(" is not allowed in a `@safe` function");
+            else
+                buf.writestring(" is not allowed in a function with default safety with `-preview=safer`");
```

Comment:
This introduced a new case for this issue. https://gcc.gnu.org/bugzilla/show_bug.cgi?id=109681